### PR TITLE
NEXUS-7898 ext model update

### PR DESCRIPTION
**Pull request by kellyrob99**
https://issues.sonatype.org/browse/NEXUS-7898

Notable here for conversation: @NotBlank versus @NotEmpty for Strings to prevent pure whitespace values from slipping in (http://stackoverflow.com/questions/17137307/in-hibernate-validator-4-1-what-is-the-difference-between-notnull-notempty)

### CODE REVIEW

**Comment by kellyrob99** [comment]
:+1:  by CI http://bamboo.s/browse/NX3-OSSF191-1

File: plugins/rapture/nexus-coreui-plugin/src/main/resources/static/rapture/NX/coreui/store/AnalyticsEvent.js [hunk 1]
```diff
@@ -37,6 +37,5 @@ Ext.define('NX.coreui.store.AnalyticsEvent', {
     }
   },
 
-  sortOnLoad: true,
```
Comment (kellyrob99) [inline plugins/rapture/nexus-coreui-plugin/src/main/resources/static/rapture/NX/coreui/store/AnalyticsEvent.js]: This property is already the default so I opted to remove it. Please speak up if you have a preference in this regard. :)

File: plugins/rapture/nexus-coreui-plugin/src/main/resources/static/rapture/NX/coreui/store/LdapServer.js [hunk 2]
```diff
@@ -37,6 +37,5 @@ Ext.define('NX.coreui.store.LdapServer', {
     }
   },
 
-  sortOnLoad: true,
-  sorters: { property: 'name', direction: 'ASC' }
+  sorters: { property: 'order', direction: 'ASC' }
```
Comment (kellyrob99) [inline plugins/rapture/nexus-coreui-plugin/src/main/resources/static/rapture/NX/coreui/store/LdapServer.js]: opinionated changed based on the fact that the list has a User specified order

**Comment by jdillon** [comment]
+1, though got a lot bigger than originally intended.  question about why all the sortOnLoad was removed.

File: plugins/rapture/nexus-coreui-plugin/src/main/resources/static/rapture/NX/coreui/store/AnalyticsEvent.js (hunk 1 above)
Comment (jdillon) [inline plugins/rapture/nexus-coreui-plugin/src/main/resources/static/rapture/NX/coreui/store/AnalyticsEvent.js]: thx for clarification, no objection if this is the default

**Comment by adreghiciu** [comment]
+1
Curious why did you add all of those constraints on XO's?

**Comment by kellyrob99** [comment]
@adreghiciu 
``` Curious why did you add all of those constraints on XO's? ```
It seemed appropriate to add the annotations as I was identifying model properties where those constraints were already being applied elsewhere. It's still not 100% but IMO it's now a lot more representative of the domain.

**Pull request merged**